### PR TITLE
feat(ingest): handle multiline string coercion

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/unit/config/test_pydantic_validators.py [hunk 1]
```diff
@@ -92,11 +97,38 @@ class TestModel(ConfigModel):
...
     assert v.b == "original"
     assert v.d1 == "deprecated"
     assert v.d2 == "deprecated"
     assert any(["d1 is deprecated" in warning for warning in get_global_warnings()])
     assert any(["d2 is deprecated" in warning for warning in get_global_warnings()])
 
     clear_global_warnings()
+
+
+def test_multiline_string_fixer():
```
Comment (asikowitz) [inline metadata-ingestion/tests/unit/config/test_pydantic_validators.py:113]: Can you add a test with a dash in the secret name?

**Review by asikowitz — APPROVED** [review]

File: docs/developers.md [hunk 2]
```diff
@@ -17,10 +17,8 @@ title: "Local Development"
 On macOS, these can be installed using [Homebrew](https://brew.sh/).
 
 ```shell
-# Install Java 8 and 11
-brew tap homebrew/cask-versions
-brew install java11
-brew install --cask zulu8
+# Install Java
+brew install openjdk@17
```
[the remark this answers is not in the document]
Comment (hsheth2) [reply docs/developers.md:21]: it should be fine as long as it's java 17

File: docs-website/download_historical_versions.py [hunk 3]
```diff
@@ -37,9 +37,9 @@ def fetch_urls(
         except Exception as e:
             if attempt < max_retries:
                 print(f"Attempt {attempt + 1}/{max_retries}: {e}")
-                time.sleep(retry_delay)
+                time.sleep(retry_delay * 2**attempt)
```
[the remark this answers is not in the document]
Comment (hsheth2) [reply docs-website/download_historical_versions.py:40]: We didn't want to set up a venv here, so we're using pure python without any external packages

File: metadata-ingestion/tests/unit/config/test_pydantic_validators.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/tests/unit/config/test_pydantic_validators.py:113]: that will come in a follow up PR - this one doesn't fix that just yet

**Pull request merged**